### PR TITLE
Run initialization on ODEs with superset initial conditions

### PR DESCRIPTION
**Pull request by ChrisRackauckas**
InitializationProblem is not always built, however it missed a case. If you have an ODE and you define all of the initial conditions, then it does not need an initialization problem. But, if you simplify down to an ODE and give initial conditions for all of the original variables, including some simplified out, then you have defined initial conditions on some observed quantities and that needs to be accounted for by an initialization process.

This fixes https://github.com/SciML/ModelingToolkit.jl/issues/2619

### CODE REVIEW

**Comment by ChrisRackauckas** [comment]
@AayushSabharwal did some of the cases of initialization with array variables simply require it wasn't using the initialization?

File: src/systems/diffeqs/abstractodesystem.jl [hunk 1]
```diff
@@ -777,6 +777,7 @@ function process_DEProblem(constructor, sys::AbstractODESystem, u0map, parammap;
     varmap = canonicalize_varmap(varmap)
     varlist = collect(map(unwrap, dvs))
     missingvars = setdiff(varlist, collect(keys(varmap)))
+    setboserved = setdiff(collect(keys(varmap)), varlist)
```
Comment (AayushSabharwal) [inline src/systems/diffeqs/abstractodesystem.jl:780]:
```suggestion
    setobserved = setdiff(collect(keys(varmap)), varlist)
```

File: src/systems/diffeqs/abstractodesystem.jl [hunk 2]
```diff
@@ -824,7 +825,7 @@ function process_DEProblem(constructor, sys::AbstractODESystem, u0map, parammap;
     # TODO: make it work with clocks
     # ModelingToolkit.get_tearing_state(sys) !== nothing => Requires structural_simplify first
     if sys isa ODESystem && build_initializeprob &&
-       (((implicit_dae || !isempty(missingvars)) &&
+       (((implicit_dae || !isempty(missingvars) || !isempty(setboserved)) &&
```
Comment (AayushSabharwal) [inline src/systems/diffeqs/abstractodesystem.jl:828]:
```suggestion
       (((implicit_dae || !isempty(missingvars) || !isempty(setobserved)) &&
```

**Review by AayushSabharwal — APPROVED** [review]
Just a typo

File: src/systems/diffeqs/abstractodesystem.jl [hunk 3]
```diff
@@ -810,7 +811,8 @@ function process_DEProblem(constructor, sys::AbstractODESystem, u0map, parammap;
 
     # ModelingToolkit.get_tearing_state(sys) !== nothing => Requires structural_simplify first
     if sys isa ODESystem && build_initializeprob &&
-       (((implicit_dae || !isempty(missingvars)) &&
+       (((implicit_dae || !isempty(missingvars) || !isempty(setobserved)) &&
+         all(isequal(Continuous()), ci.var_domain) &&
```
Comment (ChrisRackauckas) [inline src/systems/diffeqs/abstractodesystem.jl:815]:
```suggestion
       (((implicit_dae || !isempty(missingvars) || !isempty(setobserved)) &&
```

**Pull request closed without merging**